app/services/doc_auth/error_generator.rb

```diff
@@ -252,7 +151,182 @@ def get_selfie_error(liveness_enabled, response_info)
+      'Document Expired': { type: ID, msg_key: Errors::DOCUMENT_EXPIRED_CHECK },
+      'Document Number Crosscheck': { type: ID, msg_key: Errors::DOC_NUMBER_CHECKS },
+      'Expiration Date Crosscheck': { type: ID, msg_key: Errors::EXPIRATION_CHECKS },
+      'Expiration Date Valid': { type: ID, msg_key: Errors::EXPIRATION_CHECKS },
+      'Full Name Crosscheck': { type: ID, msg_key: Errors::FULL_NAME_CHECK },
+      'Issue Date Crosscheck': { type: ID, msg_key: Errors::ISSUE_DATE_CHECKS },
+      'Issue Date Valid': { type: ID, msg_key: Errors::ISSUE_DATE_CHECKS },
+      'Layout Valid': { type: ID, msg_key: Errors::ID_NOT_VERIFIED },
+      'Near-Infrared Response': { type: ID, msg_key: Errors::ID_NOT_VERIFIED },
+      'Photo Printing': { type: FRONT, msg_key: Errors::VISIBLE_PHOTO_CHECK },
+      'Physical Document Presence': { type: ID, msg_key: Errors::ID_NOT_VERIFIED },
+      'Sex Crosscheck': { type: ID, msg_key: Errors::SEX_CHECK },
+      'Visible Color Response': { type: ID, msg_key: Errors::VISIBLE_COLOR_CHECK },
+      'Visible Pattern': { type: ID, msg_key: Errors::ID_NOT_VERIFIED },
+      'Visible Photo Characteristics': { type: FRONT, msg_key: Errors::VISIBLE_PHOTO_CHECK },
+    }.freeze
+
+    SUPPORTED_ID_CLASSNAME = ['Identification Card', 'Drivers License'].freeze
+
+    def initialize(config)
+      @config = config
+    end
+
+    def generate_doc_auth_errors(response_info)
+      # when entered here, it's decided the doc auth is not successful
+
+      # scan unknown(handled) error, make sure `warn_notify` it
+      # if unhandled error found
+      unknown_fail_count = scan_for_unknown_alerts(response_info)
+
+      # check whether ID type supported
+      id_type_error = IdTypeErrorHandler.new.handle(response_info)
+      return id_type_error.to_h if id_type_error.present? && !id_type_error.empty?
+
+      # check Image metrics error
+      metrics_error = ImageMetricsErrorHandler.new(config).handle(response_info)
+      return metrics_error.to_h if metrics_error.present? && !metrics_error.empty?
+
+      # check selfie error
+      selfie_error_handler = SelfieErrorHandler.new
+      selfie_error = selfie_error_handler.handle(response_info)
+
+      # if selfie itself is ok, but we have selfie related error
+      if selfie_error_handler.present_generic_selfie_error?(selfie_error)
+        return SelfieErrorHandler::SELFIE_GENERAL_FAILURE_ERROR
+      end
+
+      # other vendor response detail error
+      liveness_enabled = response_info[:liveness_enabled]
+      alert_error_count = response_info[:doc_auth_result] == 'Passed' ?
+                            0 : response_info[:alert_failure_count]
+
+      known_alert_error_count = alert_error_count - unknown_fail_count
+      alert_error = AlertErrorHandler.new(config: config, liveness_enabled: liveness_enabled).
+        handle(known_alert_error_count, response_info, selfie_error)
```

Comment:
```suggestion
      alert_error_handler = AlertErrorHandler.new(config: config, liveness_enabled: liveness_enabled)
      alert_error = alert_error_handler.handle(known_alert_error_count, response_info, selfie_error)
```

i think this would be more debuggable 🤔